org.jacoco.core.test/src/org/jacoco/core/test/validation/Source.java

```diff
@@ -44,6 +44,21 @@ public static Source getSourceFor(final String srcFolder,
 			final Class<?> type) throws IOException {
 		File folder = new File(srcFolder);
 		File file = new File(folder, type.getName().replace('.', '/') + ".java");
+
+		// Check for existence of Kotlin source files
+		if (!file.exists()) {
+			// Remove "Kt" suffix for Kotlin classes
+			String path = type.getName().replace('.', '/');
+			if (path.endsWith("Kt")) {
```

Comment:
@Godin Ups, that looks a bit too pragmatic for me. Can't we use `IClassCoverage.getSourceFileName()` of our own API for this?

Reply:
@marchof AFAIR currently `IClassCoverage.getSourceFileName()` is not available at a time of construction of `Source`.

Reply:
@Godin Sure, but probably this can be easily reworked.